File: ks.py
class Item:
  def __init__(self,value,weight):
    self.value = value
    self.weight = weight


def knap(arr,w):
  arr.sort(key=lambda x:x.value/x.weight, reverse=True)
  finalValue = 0.0
  for i in arr:
    if (i.weight<=w):
      w-=i.weight
      finalValue+= i.value
    else:
      finalValue += (w/i.weight)*i.value
      break
  return finalValue



if __name__ == '__main__':
  w = 50
  arr = [Item(60,10),Item(100,20),Item(120,30)]
  knap1 =  knap(arr,w)
  print(knap1)

File: ks.cpp
#include<bits/stdc++.h>
using namespace std;

struct Item{
    int value;
    int weight;
};

bool cmp(Item &a,Item &b){
    return (a.value/ (double)a.weight) > (b.value/ (double)b.weight);
}

double fractionalKnapsack(Item *items,int W,int n){
    sort(items,items+n,cmp);
    double mx=0;
    for(int i=0;i<n;i++){
        if(items[i].weight<=W){
            mx+=items[i].value;
            W-=items[i].weight;
        }
        else{
            mx+=(items[i].value/(double)items[i].weight)*(double)W;
            break;
        }
    }
    return mx;
}
int main(){
    int n,W;
    cin>>n>>W;
    Item items[n];
    for(int i=0;i<n;i++){
        cin>>items[i].value;
        cin>>items[i].weight;
    }
    double maxValue=fractionalKnapsack(items,W,n);
    cout<<maxValue<<endl;
    return 0;
}
